src/vs/workbench/api/common/extHost.protocol.ts

```diff
@@ -1742,8 +1742,8 @@ export interface ExtHostCommentsShape {
 }
 
 export interface INotebookSelectionChangeEvent {
-	// handles
-	selections: number[];
+	primary: number | null;
+	selections: ICellRange[];
```

Comment:
When is primary `null`? Would it makes more sense to say that primary is just `selections[0]`?

Reply:
I thought about whether `primary` is needed at all when reading 

```
// @rebornix
// todo@API should replace selection
// never empty!
// primary/secondary selections
readonly selections: NotebookCellRange[];
```

but realized that when users deleted all cells in a document, `selections` is actually `[]` and the primary selection is `null | undefined`.

We can make `primary === selections[0]` always in the UI to make it easier to use. The work left is then making sure it's translated to `ListView.selections/focus` correctly.

Reply:
Internally we talk about `selections: ICellRange[]` between main and ext host. The only exception is `applyEdit` which still allows you to set end selections based on handles.